homeassistant/helpers/entity.py

```diff
@@ -514,6 +514,10 @@ def _async_write_ha_state(self) -> None:
         if self.available:
             attr.update(self.state_attributes or {})
             extra_state_attributes = self.extra_state_attributes
+            # Backwards compatibility for "device_state_attributes" deprecated in 2021.4
+            # Add warning in 2021.6, remove in 2021.10
```

Comment:
Let's update the comments + add a warning print
```suggestion
            # Backwards compatibility for "device_state_attributes" deprecated in 2021.4
            # Warning added in 2021.12, will be removed in 2021.4
            report_issue = self._suggest_report_issue()
            _LOGGER.warning(
                "Entity %s (%s) implements device_state_attributes. Please %s",
                self.entity_id,
                type(self),
                report_issue,
            )
```